src/mlpack/core/data/load_categorical_csv.hpp

```diff
@@ -0,0 +1,342 @@
+            tok += token;
+          }
+          token = tok;
+        }
+        info.template MapFirstPass<T>(std::move(token), rows - 1);
+      }
+    }
+  } 
+}
+
+template<typename T, typename PolicyType>
+void LoadCSV::TransposeParse(arma::Mat<T>& inout, DatasetMapper<PolicyType>& infoSet)
+{
+  // Get matrix size.  This also initializes infoSet correctly.
+  size_t rows, cols;
+  InitializeTransposeMapper<T>(rows, cols, infoSet);
+
+  // Set the matrix size.
+  inout.set_size(rows, cols);
+
+  // Initialize auxiliary variables.
+  size_t row = 0;
+  size_t col = 0;
+  std::string line;
+  inFile.clear();
+  inFile.seekg(0, std::ios::beg);
+
+  while (std::getline(inFile, line))
+  {
+    // Remove whitespaces from either side
+    trim(line);
+    // Reset the row we are looking at.  (Remember this is transposed.)
+    row = 0;
+    std::stringstream lineStream;
+    std::string token;
+    lineStream.clear();
+    lineStream.str(line);
+
+    while (lineStream.good())
+    {
+      std::getline(lineStream, token, delim);
+      // Remove whitespaces from either side
+      trim(token);
+
+      if (token[0] == '"' && token[token.size() - 1] != '"')
+      {
+        // first part of the string
+        std::string tok = token;
+        while (token[token.size() - 1] != '"')
+        {
+          tok += delim;
+          std::getline(lineStream, token, delim);
+          tok += token;
+        }
+        token = tok;
+      }
+      inout(row, col) = infoSet.template MapString<T>(std::move(token), row);
+      row++;
+    }
+
+    // Make sure we got the right number of rows.
+    if (row != rows)
+    {
+      std::ostringstream oss;
+      oss << "LoadCSV::TransposeParse(): wrong number of dimensions (" << row
+      << ") on line " << col << "; should be " << rows << " dimensions.";
+      throw std::runtime_error(oss.str());
+    }
+    // Increment the column index.
+    ++col;
+  }
+}
+
+template<typename T, typename PolicyType>
+void LoadCSV::NonTransposeParse(arma::Mat<T>& inout,
+                       DatasetMapper<PolicyType>& infoSet)
+{
+  // Get the size of the matrix.
+  size_t rows, cols;
+  InitializeMapper<T>(rows, cols, infoSet);
+
+  // Set up output matrix.
+  inout.set_size(rows, cols);
+  size_t row = 0;
+  size_t col = 0;
+
+  // Reset file position.
+  std::string line;
+  inFile.clear();
+  inFile.seekg(0, std::ios::beg);
+
+  while (std::getline(inFile, line))
+  {
+    // Remove whitespaces from either side
+    trim(line);
+
+    std::stringstream lineStream;
+    std::string token;
+
+    lineStream.clear();
+    lineStream.str(line);
+
+    while (lineStream.good())
+    {
+      if (token == "\t")
+        token.clear();
+
+      std::getline(lineStream, token, delim);
+      // Remove whitespace from either side
+      trim(token);
+
+      if (token[0] == '"' && token[token.size() - 1] != '"')
+      {
+        std::string tok = token;
+        while (token[token.size() - 1] != '"')
```

Comment:
I am not sure if the indentation is wrong here, would you check please?

Reply:
Same here, it is showing correctly in my editor.
Any idea why this is happening?